packages/cli/src/ceramic-logger-plugins.ts

```diff
@@ -86,28 +113,125 @@ export class LogToFiles {
 
             if (match !== null) {
                 const docId = match[0]
-                const filePath = filePrefix + '-docids.log'
-                LogToFiles._writeStream(filePath, docId, 'w')
+                const filePath = `${basePath}${filePrefix}-docids.log`
+                const result = await LogToFiles._writeStream(blockedFiles, filePath, docId, 'w')
+                if (result.blocked) await LogToFiles._writeBlockedWarning(blockedFiles, basePath, filePath)
             }
         }
     }
 
     /**
-     * Returns true if it has been `MINUTES_TO_EXPIRATION` minutes since the file was created
-     * @notice Returns false if file is not found.
-     * Returns true if unable to get file creation datetime for some other reason.
+     * Logs and writes warning message about `filePath` being blocked.
+     * @notice If the file that this tries to write to itself is blocked,
+     * its message will be dropped and only appear in the console
+     * @param basePath Base directory for file to write to
+     * @param filePath Path to file that is blocked
+     */
+    private static async _writeBlockedWarning (
+        blockedFiles: BlockedFiles,
+        basePath: string,
+        filePath: string
+    ): Promise<void> {
+        const loggerFilePath = basePath + 'logger.log'
+        const message = `Stream blocked for file: ${filePath}\n`
+        await LogToFiles._writeStream(blockedFiles, loggerFilePath, message, 'a')
+    }
+
+    /**
+      * Writes `message` to a file and returns whether it is blocked or not
+      * @notice If the file is currently being written to by an open stream or if
+      * its buffer is full, the call to this method will return without writing
+      * @param filePath Full path of file to write to
+      * @param message Message to write to `filePath`
+      * @param writeFlag Specifies writing method (e.g. "a" for append, "w" for overwrite)
+      * @returns True if the stream is blocked and false otherwise
+      */
+    private static async _writeStream (
+        blockedFiles: BlockedFiles,
+        filePath: string,
+        message: string,
+        writeFlag: string
+    ): Promise<WriteResult> {
+        if (blockedFiles[filePath]) {
+            console.warn(`Stream busy for ${filePath}. Some logs may be dropped.`)
+            return { blocked: blockedFiles[filePath], filePath }
+        }
+
+        blockedFiles[filePath] = true
+
+        const stream = fs.createWriteStream(filePath, { flags: writeFlag })
+        stream.on('error', (err) => {
+            console.warn(err)
+            return
+        })
+        stream.on('drain', () => {
+            blockedFiles[filePath] = false
+            return
+        })
+        stream.on('finish', () => {
+            blockedFiles[filePath] = false
+            return
+        })
+        blockedFiles[filePath] = !stream.write(util.format(message) + '\n', () => {
```

Comment:
If stream.write() returns false, is that definitely because the stream is blocked (the file system write buffer is full)?  Or do we need to check a return code or something to see why exactly the write failed?

Reply:
No it is because stream is blocked and we need to wait for it to emit drain before writing again.

Reply:
okay great!  I assume it throws if there's another type of error?  What happens in that case?  I think aborting the process in the case of a logging error is probably fine and the way to go?

Reply:
Yeah i think so actually. will add that to the on error event.